### ISSUE TITLE
Prompt & input remain on screen after hitting Ctrl-C

### ISSUE THREAD

**Issue by jwodder** [zyansheep/rustyline-async#11]
When a line of input is terminated by pressing Ctrl-C, the prompt and any text entered so far will remain on the screen even as new lines are printed below them.  This is unlike lines terminated by pressing Enter, for which the line and prompt disappear, and it leads to a messy-looking interface.

Program that can be used to reproduce this behavior:

```rust
use rustyline_async::{Readline, ReadlineError};
use std::io::Write;
use std::time::Duration;
use tokio::time::sleep;

#[tokio::main]
async fn main() -> Result<(), ReadlineError> {
    let (mut rl, mut stdout) = Readline::new("prompt> ".into())?;
    loop {
        tokio::select! {
            _ = sleep(Duration::from_secs(1)) => {
                writeln!(stdout, "Message received!")?;
            }
            cmd = rl.readline() => match cmd {
                Ok(line) => {
                    writeln!(stdout, "You entered: {line:?}")?;
                    rl.add_history_entry(line.clone());
                    if line == "quit" {
                        break;
                    }
                }
                Err(ReadlineError::Eof) => {
                    writeln!(stdout, "<EOF>")?;
                    break;
                }
                Err(ReadlineError::Interrupted) => {writeln!(stdout, "^C")?; continue; }
                Err(e) => {
                    writeln!(stdout, "Error: {e:?}")?;
                    break;
                }
            }
        }
    }
    rl.flush()?;
    Ok(())
}
```

Recording of this behavior:

![cancel-bug](https://user-images.githubusercontent.com/98207/203182408-edf407d0-f786-4f6c-afbe-c5ba65cb68ac.gif)

(Incidentally, if no text is printed in response to the Ctrl-C, things get a bit messed up on the next output; perhaps that should be a separate issue.)

**Comment by zyansheep** [comment]
Good catch! It makes a lot more sense for the line to stay when you press <Enter> so you can see what commands you've sent previously.

**Comment by zyansheep** [comment]
I added a fix in the main branch. Take a look if you would like. The Ctrl-C issue might be a bit more annoying to fix tho...

**Comment by jwodder** [comment]
Sorry for not getting back to this sooner.  I've tried out the code in the OP with rustyline-async 0.3.1 and crossterm 0.26.1, but the behavior seems to be the same.

**Comment by zyansheep** [comment]
I made it so that lines stay on screen if enter is pressed to be consistent with ctrl-c. Is this not the behavior you see?

**Comment by jwodder** [comment]
Oh, I see what you mean.  That's actually the opposite of the behavior I want, as in [the program I'm writing](https://github.com/jwodder/confab), I already output what the user just entered, and sometimes it's with a timestamp prepended, so just getting rid of the output wouldn't get the results I want.  Could you add an option for whether to leave input lines on the screen?

**Comment by zyansheep** [comment]
Alright, I added two config options, one for enter and one for ctrl-c. There is some weird rendering behavior when switching between them in the middle of a readline, but it seems to work if you set it at the beginning. I don't have a lot of time to work on this project though, so if further changes are required, a PR would be much appreciated :)

**Comment by jwodder** [comment]
Thanks, that works!  I submitted #13 to add a doc comment for the new method.

**Comment by zyansheep** [comment]
Cool, I will close this when I get around to doing a cargo publish

**Comment by zyansheep** [comment]
Fixed in Release 0.3.2 :)
